admin/assets/javascripts/pupil-filter-name.js

```diff
@@ -1,43 +1,57 @@
 /**
  * Filtering pupils by name.
  */
+
+if (!window.MTCAdmin) {
+  window.MTCAdmin = {}
+}
+
 /* global $ stickyBanner */
-$(function () {
-  if ($('.filter-name').length > 0) {
-    $('#search-name').on('change keyup', function () {
-      var input = $.trim($(this).val()).toLowerCase()
-      var selAllTr = '#generatePins > tbody > tr'
-      if (input.length === 0) {
+(function () {
+  window.MTCAdmin.pupilFilter = function () {
+    if ($('.filter-name').length > 0) {
```

Comment:
how does this perform on larger sets on the first couple of inputs?  might be worth checking and only invoking after 2nd or 3rd character.

Reply:
@GuyHarwood that's existing logic from Dan's filter. I haven't notice any performance impact so far.